scss/_patterns_logo-section.scss

```diff
@@ -0,0 +1,173 @@
+      // 100% / $column-count - $gutters-total-width / $column-count
+      // Thus the calculation is split into 2 parts, which are then used in the final calc() function:
+
+      // calculate the width of a single column by dividing the entire width by the number of columns it represents:
+      $percentage-width-of-one-col: 100% / $column-count;
+
+      // multiply the width of one column by the number of columns the logo needs to span
+      $percentage-contribution: $logo-column-span * $percentage-width-of-one-col;
+
+      // calculate the number of gutters the logo spans
+      $gutter-width-inside-logo: $gutter * ($logo-column-span - 1);
+
+      // add the width of gutters spanned by the logo, subtract the $gutters-total-width / $column-count;
+      // the signs here are reversed because the variable is subtracted. This avoids having + and - in the final css, e.g.: "+ -.75rem"
+      $rem-contribution--legacy-browsers: -$gutter-width-inside-logo;
+      $rem-contribution: $logo-column-span * ($gutters-total-width / $column-count) - $gutter-width-inside-logo;
+
+      .#{$grid-column-prefix}#{$column-count} & {
+        @include vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution--legacy-browsers);
+
+        @supports (gap: 1rem) {
+          @include vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution);
+        }
+      }
+    }
+  }
+}
+
+@mixin vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution) {
+  @if $rem-contribution == 0 {
+    width: $percentage-contribution;
+  } @else {
+    width: calc(#{$percentage-contribution} - #{$rem-contribution});
+  }
+}
+
+@mixin vf-p-logo-section {
+  // The logo section pattern needs to align logos to the grid when they are an even number, and center them (while maintaining the same logo width) when the number of logos on a line is odd.
+  // This means we can't use css grid for this, instead we need to replicate its behavior for even numbers of logos, and center the odd ones.
+  .p-logo-section__title {
+    @extend %x-small-text;
+
+    font-weight: $font-weight-bold;
+    text-transform: uppercase;
+  }
+
+  .p-logo-section__items {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 0 map-get($grid-gutter-widths, small);
+    justify-content: center;
+    margin-bottom: $spv-outer--scaleable;
+    width: 100%;
+
+    @media (min-width: $threshold-4-6-col) {
+      @supports (gap: 1rem) {
+        gap: 0 map-get($grid-gutter-widths, medium);
+      }
+    }
+
+    @media (min-width: $threshold-6-12-col) {
+      @supports (gap: 1rem) {
+        gap: 0 map-get($grid-gutter-widths, large);
+      }
+    }
+  }
+
+  .p-logo-section__item {
+    align-items: center;
+    display: flex;
+    flex: 0 1 auto;
+
+    & + & {
+      margin-left: map-get($grid-gutter-widths, small);
+
+      @supports (gap: 1rem) {
```

Comment:
Unfortunately it's impossible to reliably detect `gap` supports in this case. It forks fine in IE case, because it doesn't support `gap` at all, but Safari does support `gap` in grid, but doesn't in `flex`. So Safari will pass this and will remove the margins.

<img width="742" alt="Screenshot 2021-04-09 at 13 57 23" src="https://user-images.githubusercontent.com/83575/114176520-88ab9b00-993b-11eb-9458-d58cb8dca3a1.png">

And it seems there is currently no way in CSS to detect support for gap in context of flex layout.

So we have to either accept the lack or margins in Safari, or use margins everywhere and drop `gap` here completely.